src/wallet/wallet.cpp

```diff
@@ -2492,9 +2492,10 @@ CCoinsViewCache::outputMaturity CWalletObjBase::IsOutputMature(unsigned int vOut
 
     if (IsCertificate())
     {
-        // we do not consider even an unconfirmed change output in a certificate as mature
-        if ((nDepth == 0))
-            return CCoinsViewCache::outputMaturity::IMMATURE;
+        // we consider an unconfirmed change output in a certificate as mature
+        // even if that can only be spent by a certificate (not by a transaction)
+        if ((nDepth == 0) && !GetVout()[vOutPos].isFromBackwardTransfer)
```

Comment:
probably best checking for availability of vOutPos entry

Reply:
Another point. IsOutputMature is widely use for both balance calculations and actual search of unspent utxos. If you choose to declare change output mature since it may be used for financing certificates, you need then to crosscheck, that caller use this utxo to finance only a certificate and not a tx.